src/Generation/ServiceDetails.php

```diff
@@ -114,16 +115,23 @@ class ServiceDetails
     /** @var bool *Readonly* Whether the service is deprecated. */
     public bool $isDeprecated = false;
 
+    /**
+     * @var int *Readonly* The transport type supported in this client. Valid options are in Transport.php.
```

Comment:
if you want this to be readonly, I suggest using getters instead of public properties. 

I wouldn't mind a refactor of all these properties to use public getters without setters for this reason, as they're all labeled Readonly. Unless something else is meant by `Readonly`?

Reply:
These patterns were in place before I took on this codebase; the `readonly` addition was intended to follow the existing structure. Happy to address these in general in separate PRs, but I'd like to keep this as-is for consistency here.

If you have general structural improvement suggestions, please let me know or file issues in this repo!  I'd be happy to tackle them over time. :)